interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+                 charge=None, size=1.0, standardState=None):
+        """
+        :param name:
+            The species name (or formula). The name may be arbitrarily long,
+            although usually a relatively short, abbreviated name is most
+            convenient. Required parameter.
+        :param atoms:
+            The atomic composition, specified by a string containing
+            space-delimited ``<element>:<atoms>`` pairs. The number of atoms may be
+            either an integer or a floating-point number.
+        :param note:
+            A user-defined comment. Not evaluated by Cantera itself.
+        :param thermo:
+            The parameterization to use to compute the reference-state
+            thermodynamic properties. This must be one of the entry types
+            described in `Thermodynamic Property Models
+            <https://cantera.org/science/science-species.html#sec-thermo-models>`_.
+            To specify multiple parameterizations, each for a different temperature range,
+            group them in parentheses.
+        :param transport:
+            An entry specifying parameters to compute this species'
+            contribution to the transport properties. This must be one of the
+            entry types described in `Species Transport Coefficients
+            <https://cantera.org/science/science-species.html#species-transport-coefficients>`_,
+            and must be consistent with the transport model of the phase into which
+            the species is imported. To specify parameters for multiple
+            transport models, group the entries in parentheses.
+        :param size:
+            The species "size". Currently used only for surface species,
+            where it represents the number of sites occupied.
+        :param charge:
+            The charge, in multiples of :math:`|e|`. If not specified, the
+            charge will be calculated from the number of "atoms" of element
+            ``E``, which represents an electron.
+        """
+        self._name = name
+        self._atoms = getComposition(atoms)
+        if charge is not None and 'E' not in self._atoms:
+            self._atoms['E'] = -charge
+        self._size = size
+        self._comment = note
+
+        if isinstance(thermo, (list, tuple)):
+            if isinstance(thermo[0], (NASA, NASA9, Shomate)):
+                self._thermo = MultiPolyThermo(thermo)
+        elif isinstance(thermo, (NASA, NASA9, Shomate)):
+            self._thermo = MultiPolyThermo([thermo])
+        elif thermo is not None:
+            self._thermo = thermo
+        else:
+            self._thermo = const_cp()
+
+        self._transport = transport
+        self._standardState = standardState
+
+        self.rk_pure = {}
+        self.rk_binary = {}
+        self.density = None
+
+        _species.append(self)
+        _speciesnames.append(name)
+
+    @classmethod
+    def to_yaml(cls, representer, node):
```

Comment:
Don't ask me why it works this way -- this is the interface required by [`ruamel.yaml`](https://yaml.readthedocs.io/en/latest/dumpcls.html). It makes even less sense when you realize that it ought to just be a regular member function with the signature `to_yaml(self, representer)`, because `node` is an instance of this class.

Reply:
From the example, it looks like it expects the `yaml_tag` to be a class variable, instead of an instance variable. Why that's necessary, I don't know. Oy.